crates/sui-graphql-rpc/src/types/balance.rs

```diff
@@ -249,3 +274,24 @@ fn filter(mut query: RawQuery, owner: SuiAddress, coin_type: Option<TypeTag>) ->
 
     query
 }
+
+/// Check that the cursors, if provided, have the same checkpoint_viewed_at.
+pub(crate) fn validate_cursor_consistency(
```

Comment:
Worth thinking about a neat way to abstract this, seeing as it's quite a common pattern. Sketch for that:

- Add a trait: `ConsistentCursor: CursorType` that exposes the `checkpoint_viewed_at`.
- Split `Target<C>` into `Target<C>` (as it was before) and `ConsistentTarget<C: ConsistentCursor>: Target<C>` that exposes the `consistent_cursor` builder.
- Add an impl block: `impl<C: ConsistentCursor> Page<C> { fn validate_consistency(&self) -> Option<u64> }`.

Reply:
Applied these concepts later in the stack at https://github.com/MystenLabs/sui/pull/15996